### PR TITLE
Don't mark non-static methods as static.

### PR DESCRIPTION
**Pull request by fdamken**

### CODE REVIEW

File: Pyrado/pyrado/utils/experiments.py [hunk 1]
```diff
@@ -67,24 +74,176 @@ def load_experiment(
...
     # Hyper-parameters
     extra["hparams"] = load_hyperparameters(ex_dir)
 
-    # Algorithm, environment, policy, and more
-    algo = pyrado.load("algo.pkl", ex_dir)
-    env, policy, extra = algo.load_snapshot(args)
+    # Algorithm specific
+    algo = Algorithm.load_snapshot(load_dir=ex_dir, load_name="algo")
+
+    if algo.name == "spota":
```
Comment (miterion) [inline Pyrado/pyrado/utils/experiments.py:91]: This is weird, why did you add it again?

File: Pyrado/pyrado/utils/experiments.py [hunk 2]
```diff
@@ -67,24 +74,176 @@ def load_experiment(
...
+        extra["teacher_critics"] = algo.teacher_critics
+        extra["teacher_ex_dirs"] = algo.teacher_ex_dirs
+
+    else:
+        raise pyrado.TypeErr(msg="No matching algorithm name found during loading the experiment!")
 
     # Check if the return types are correct. They can be None, too.
-    if env is not None and not isinstance(env, (Env, EnvWrapper)):
-        raise pyrado.TypeErr(given=env, expected_type=[Env, EnvWrapper])
+    if env is not None and not isinstance(env, (SimEnv, EnvWrapper)):
```
Comment (miterion) [inline Pyrado/pyrado/utils/experiments.py:245]: Sure about this change? Could this lead to conflicts with real envs?

File: Pyrado/pyrado/utils/experiments.py [hunk 3]
```diff
@@ -202,8 +360,6 @@ def cpp_export(
                               This requires the experiment's XML configuration to be exported beforehand.
     :param policy_node_name: name of the control policies node in the XML file, e.g. 'policy' or 'preStrikePolicy'
     """
-    from pyrado.environments.rcspysim.base import RcsSim
```
Comment (miterion) [inline Pyrado/pyrado/utils/experiments.py:205]: I do not like inline imports but as someone who does not use the RcsSim (nor have it installed) putting this line in the imports breaks simulacra.

**Review by miterion — CHANGES_REQUESTED** [review]
Personally I am not a fan of this style since `@staticmethod` is already handling this well, but it does not hurt either. There are however a few problems introduced by this PR

File: Pyrado/pyrado/utils/experiments.py (hunk 1 above)
Comment (fdamken) [reply Pyrado/pyrado/utils/experiments.py]: Merging error…

File: Pyrado/pyrado/utils/experiments.py (hunk 2 above)
Comment (fdamken) [reply Pyrado/pyrado/utils/experiments.py]: Merging error…

File: Pyrado/pyrado/utils/experiments.py (hunk 3 above)
Comment (fdamken) [reply Pyrado/pyrado/utils/experiments.py]: Merging error…

**Review by famura — APPROVED** [review]

**Comment by famura** [comment]
May I ask why you want this change, i.e., what is the benefit?

**Comment by fdamken** [comment]
> May I ask why you want this change, i.e., what is the benefit?

The main benefit is e.g. in the DomainParamTransformation class where the methods where marked as static, but are used as instance methods, including overriding. The other things are just changes in notation, there is no big benefit, just consistency.

**Review by miterion — APPROVED** [review]

**Comment by famura** [comment]
Consistency is big enough for me :D
